[Unified Search] Hide ES|QL ad hoc data views from the data view picker

davismcphee opened the pull request:
## Summary

This PR uses the new `esql` data view `type` to hide ES|QL ad hoc data views from the Unified Search data view picker since they are an implementation detail and should not be visible to users.

Resolves #170098.
Resolves #166911.
Partially addresses #176873.

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

davismcphee commented:
/ci

davismcphee commented:
/ci

davismcphee commented:
/ci

davismcphee commented on src/plugins/unified_search/public/dataview_picker/data_view_picker.tsx:
```diff
@@ -18,7 +18,6 @@ export type ChangeDataViewTriggerProps = EuiButtonProps & {
 };
 
 export enum TextBasedLanguages {
-  SQL = 'SQL',
```
I dropped this since we don't support SQL anymore and it was unused outside of tests.

stratoula commented on src/plugins/unified_search/public/dataview_picker/change_dataview.test.tsx:
```diff
@@ -165,14 +165,16 @@ describe('DataView component', () => {
-          textBasedLanguages: [TextBasedLanguages.ESQL, TextBasedLanguages.SQL],
-          textBasedLanguage: TextBasedLanguages.ESQL,
+          ...(esqlMode && {
+            textBasedLanguages: [TextBasedLanguages.ESQL, TextBasedLanguages.ESQL],
```
```suggestion
            textBasedLanguages: [TextBasedLanguages.ESQL],
```

davismcphee replied:
I blame find and replace 🙂 Fixed in this commit: https://github.com/elastic/kibana/pull/177109/commits/fccf8a70dc7707dfa388241008269c7c4840ffdd.

stratoula reviewed (approved):
Changes LGTM, just a minor comment. Thanx Davis!

jughosta commented:
Should we also hide it from Alerts data view picker?

<img width="500" alt="Screenshot 2024-02-20 at 09 37 36" src="https://github.com/elastic/kibana/assets/1415710/68add429-039d-4efb-adcd-ad7a71079a70">

davismcphee commented:
>Should we also hide it from Alerts data view picker?

@jughosta Good catch! I updated the approach to do the filtering within the `DataViewsList` component used by both the data view picker and the alerts flyout. Doing it this way also means other consumers can get the filtering for free as long as they include `type` in their data view list items: https://github.com/elastic/kibana/pull/177109/commits/fccf8a70dc7707dfa388241008269c7c4840ffdd.

cc @stratoula in case you want to do another quick review since I changed the approach slightly.

stratoula commented:
I am not testing again but I am super ok with the new changes!

jughosta reviewed (approved):
LGTM 👍
